jgrapht-core/src/main/java/org/jgrapht/alg/cycle/ChordalityInspector.java

```diff
@@ -167,45 +239,260 @@ public boolean isPerfectEliminationOrder(List<V> vertexOrder) {
         return order;
     }
 
+    /**
+     * Lazily computes the coloring of the graph. Returns null if the graph isn't chordal.
+     *
+     * @return coloring of the graph if it is chordal, null otherwise.
+     */
+    private Coloring<V> lazyComputeColoring() {
+        if (coloring == null) {
+            if (!isComputed()) {
+                isChordal();
+            }
+            if (chordal) {
+                Map<V, Integer> vertexColoring = new HashMap<>(order.size());
+                Map<V, Integer> vertexInOrder = getVertexInOrder(order);
+                for (V vertex : order) {
+                    vertexColoring.put(vertex, getPredecessors(vertexInOrder, vertex).stream().map(vertexColoring::get)
+                            .max(Integer::compareTo).orElse(-1) + 1);
+                }
+                int maxColor = order.stream().map(vertexColoring::get).max(Integer::compareTo).orElse(-1) + 1;
+                coloring = new ColoringImpl<>(vertexColoring, maxColor);
+                return coloring;
+            } else {
+                return null;
+            }
+        } else {
+            return coloring;
+        }
+    }
+
+    /**
+     * Lazily computes a maximum independent set of the inspected {@code graph}.
+     * If the graph isn't chordal, returns null.
+     *
+     * @return a maximum independent set of the {@code graph} if it is chordal, false otherwise.
+     */
+    private Set<V> lazyComputeMaximumIndependentSet() {
+        if (maximumIndependentSet == null) {
+            if (!isComputed()) {
+                isChordal();
+            }
+            if (chordal) {
+                Set<V> restricted = new HashSet<>();
+                Set<V> independent = new HashSet<>();
+                ListIterator<V> reverse = order.listIterator(order.size());
+                while (reverse.hasPrevious()) {
+                    V previous = reverse.previous();
+                    if (!restricted.contains(previous)) {
+                        independent.add(previous);
+                        for (E edge : graph.edgesOf(previous)) {
+                            V opposite = Graphs.getOppositeVertex(graph, edge, previous);
+                            if (!previous.equals(opposite)) {
+                                restricted.add(opposite);
+                            }
+                        }
+                    }
+                }
+                return maximumIndependentSet = independent;
```

Comment:
This method can be reduced in size by removing the last 2 else clauses, and retaining the `return maximumIndependentSet` at the end.